scripts/publish.js

```diff
@@ -46,6 +47,28 @@ if (!publishers.includes('salesforce')) {
   exit(-1);
 }
 
+// Checks that you have specified the next version as an environment variable, and that it's properly formatted.
+const nextVersion = process.env['SALESFORCEDX_VSCODE_VERSION'];
+if (!nextVersion) {
+  console.log(
+    'You must specify the next version of the extension by setting SALESFORCEDX_VSCODE_VERSION as an environment variable.'
+  );
+  exit(-1);
+} else {
+  const [version, major, minor, patch] = nextVersion.match(
+    /^(\d+)\.(\d+)\.(\d+)$/
+  );
+}
+
+// Checks that a tag of the next version doesn't already exist
+const checkTags = shell.exec('git tag', { silent: true }).stdout;
+if (checkTags.includes(nextVersion)) {
+  console.log(
+    `A git tag already exists. Try running \'git tag -d v${nextVersion}\'`
```

Comment:
I would be more precise here and say you probably should reclone the whole repository and start fresh to avoid any problems. That's how an autobuild will do it and that's how we should do it.